### PR TITLE
support nesting flows

### PR DESCRIPTION
**Pull request by cpolcyn**
Allow flows to call other flows. Example usage in `cumulusci.yml`:

```
flows:
    example_flow:
        tasks:
            1:
                task: example_task
            2:
                flow: another_flow
```

- A given flow may only occur once in the tree, to prevent infinite recursion
- Nested flows inherit the project config, org config, flow options, and skipped tasks from parent flows
- Return values from nested flows can be referenced just like tasks: flow1.flow2.task.key

### CODE REVIEW

**Comment by cpolcyn** [comment]
Thinking about it some more, I think that task options should not be passed through to nested flows. They are particular to a task within a given flow and not meant to be applied to the flow as a whole.

File: cumulusci/core/flows.py [hunk 1]
```diff
@@ -27,15 +31,16 @@ def __init__(self, project_config, flow_config, org_config, options=None, skip=N
         """ A collection of result objects in task execution order """
         self.tasks = []
         """ A collection of configured task objects, either run or failed """
-        self._init_options(options)
+        self.nested = nested
```
Comment (cdcarter) [inline cumulusci/core/flows.py]: can you add a docstring? i'm  pretty sure this indicates we are on an inner flow?

File: cumulusci/core/flows.py [hunk 2]
```diff
@@ -72,26 +77,46 @@ def _init_flow(self):
...
+        if not self.nested:
+            self._check_infinite_flows(self.flow_config.tasks)
         tasks = []
         for step_num, config in list(self.flow_config.tasks.items()):
-            if config['task'] == 'None':
+            if 'flow' in config and 'task' in config:
+                raise ConfigError('"flow" and "task" in same config item')
+            if (('flow' in config and config['flow'] == 'None') or
+                ('task' in config and config['task'] == 'None')):
                 continue
```
Comment (cdcarter) [inline cumulusci/core/flows.py]: we support no-op tasks?

File: cumulusci/core/flows.py [hunk 3]
```diff
@@ -72,26 +77,46 @@ def _init_flow(self):
...
-                    'task_config': self.project_config.get_task(
-                        config['task']
-                    ),
+                    'task_config': task_config,
                 }
             ))
-        tasks.sort()
-        tasks = [task_info[1] for task_info in tasks]
+        tasks.sort()  # sort by step number
+        tasks = [task[1] for task in tasks]  # drop the sort keys
```
Comment (cdcarter) [inline cumulusci/core/flows.py]: thank you for these clarifying comments, this was magic wasn't it...

File: cumulusci/core/flows.py [hunk 4]
```diff
@@ -167,8 +224,15 @@ def _run_task(self, flow_task_config):
             if str(value).startswith('^^'):
                 value_parts = value[2:].split('.')
                 parent = self._find_task_by_name(value_parts[0])
-                for attr in value_parts[1:]:
-                    parent = parent.return_values.get(attr)
+                n = 0
+                while isinstance(parent, BaseFlow):
+                    n += 1
+                    parent = parent._find_task_by_name(value_parts[n])
+                for attr in value_parts[(n + 1):]:
```
Comment (cdcarter) [inline cumulusci/core/flows.py]: this took me a second to figure out but is really smart. nice.

File: cumulusci/core/flows.py (hunk 1 above)
Comment (cpolcyn) [reply cumulusci/core/flows.py]: Sure, you mean a docstring for `__init__`?

File: cumulusci/core/flows.py (hunk 2 above)
Comment (cpolcyn) [reply cumulusci/core/flows.py]: Hmm good catch, I suppose this should raise a config exception?

Comment (cpolcyn) [reply cumulusci/core/flows.py]: On second thought this keeps the functionality we had before (see deleted line 82) so I'll leave it like this. I think the idea is to no-op when overriding a task in another flow, like using a deploy+test flow but skipping tests.

Comment (cdcarter) [reply cumulusci/core/flows.py]: that makes a ton of sense. can you add a comment that this is to allow overrides in yaml?

File: cumulusci/core/flows.py (hunk 1 above)
Comment (cdcarter) [reply cumulusci/core/flows.py]: just like above tasks or options

see http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html and cmd-f for "self.attr5 = None" to see how it looks i guess

**Comment by cpolcyn** [comment]
@cdcarter I addressed your comments with some new commits. Please re-review.

**Pull request merged**